mvcc/index.go

```diff
@@ -87,17 +87,31 @@ func (ti *treeIndex) keyIndex(keyi *keyIndex) *keyIndex {
 
 func (ti *treeIndex) visit(key, end []byte, f func(ki *keyIndex)) {
 	keyi, endi := &keyIndex{key: key}, &keyIndex{key: end}
-
-	ti.RLock()
-	defer ti.RUnlock()
-
-	ti.tree.AscendGreaterOrEqual(keyi, func(item btree.Item) bool {
-		if len(endi.key) > 0 && !item.Less(endi) {
-			return false
-		}
-		f(item.(*keyIndex))
-		return true
-	})
+	next := true
+
+	for next {
+		i := 0
+
+		ti.RLock()
```

Comment:
@heyitsanthony This is for reducing locking time. Or read will still block write for 100ms when ranging over the index. Luckily, our index itself is MVCC. So we can release and acquire again as we wish.